pandas/core/groupby/generic.py

```diff
@@ -1181,6 +1197,63 @@ def blk_func(bvalues: ArrayLike) -> ArrayLike:
 
         return new_mgr
 
+    def _cython_agg_arrays(
+        self, how: str, alt=None, numeric_only: bool = True, min_count: int = -1
+    ) -> Tuple[List[ArrayLike], Index]:
+
+        data: ArrayManager = self._get_data_to_aggregate()
+
+        if numeric_only:
+            data = data.get_numeric_data(copy=False)
+
+        def py_fallback(bvalues: ArrayLike) -> ArrayLike:
+            # TODO(ArrayManager)
+            raise NotImplementedError
+
+        def array_func(values: ArrayLike) -> ArrayLike:
+
+            try:
+                result = self.grouper._cython_operation(
+                    "aggregate", values, how, axis=1, min_count=min_count
+                )
+            except NotImplementedError:
+                # generally if we have numeric_only=False
+                # and non-applicable functions
+                # try to python agg
+
+                if alt is None:
+                    # we cannot perform the operation
+                    # in an alternate way, exclude the block
+                    assert how == "ohlc"
+                    raise
+
+                result = py_fallback(values)
+
+            return _cast_agg_result(result, values, how, reshape1d=False)
```

Comment:
looks like reshape1d=False is the only difference from the blk_func version; can this be refactored/shared similar to cast_agg_result?

Reply:
Yes, possibly, will take a look. I initially started with a separate version, as that was easier to experiment with / adapt, but indeed might now be possible to re-factor into one method.